src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -632,6 +631,7 @@ internal bool TryRead(out ReadResult result)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static void TrySchedule(PipeScheduler scheduler, Action<object> action, object state)
         {
             if (action != null)
```

Comment:
I'm probably just missing something, but why is there a null check here at all?  It looks like the only callers to TryScheduler(scheduler, action, state) pass in s_invokeCompletionCallbacks as the action:
https://source.dot.net/#System.IO.Pipelines/System/IO/Pipelines/Pipe.cs,15511ec2233785f4